From 5b4ad9d0df45bb9704b12927c793a65ccc7a42fa Mon Sep 17 00:00:00 2001
From: Li Zhou <li.zhou@windriver.com>
Date: Thu, 7 Apr 2022 11:42:47 +0800
Subject: [PATCH 2/8] intel-iavf: pass linux common header to check_aux_bus

For debian, there are two header pkgs, e.g. linux-headers-5.10.0-6-amd64
and linux-headers-5.10.0-6-common. linux-headers-5.10.0-6-amd64 is
enough before check_aux_bus is added. But check_aux_bus need files
in linux-headers-5.10.0-6-common. So pass linux-headers-5.10.0-6-common
path as the KSRC for check_aux_bus when KSRC_COMMON is exported in
debian rules.

Signed-off-by: Li Zhou <li.zhou@windriver.com>
[jpm: Refresh patch for iavf v4.5.3)]
Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 src/common.mk | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/common.mk b/src/common.mk
index 0b2ad05..a0b65d6 100644
--- a/src/common.mk
+++ b/src/common.mk
@@ -410,9 +410,14 @@ export INSTALL_MOD_DIR ?= updates/drivers/net/ethernet/intel/${DRIVER}
 # If the check_aux_bus script exists, then this driver depends on the
 # auxiliary module. Run the script to determine if we need to include
 # auxiliary files with this build.
+ifeq (${KSRC_COMMON},)
+KSRC_AUX := ${KSRC}
+else
+KSRC_AUX := ${KSRC_COMMON}
+endif
 CHECK_AUX_BUS ?= ../scripts/check_aux_bus
 ifneq ($(call test_file,${CHECK_AUX_BUS}),)
-NEED_AUX_BUS := $(shell ${CHECK_AUX_BUS} --ksrc="${KSRC}" --build-kernel="${BUILD_KERNEL}" >/dev/null 2>&1; echo $$?)
+NEED_AUX_BUS := $(shell ${CHECK_AUX_BUS} --ksrc="${KSRC_AUX}" --build-kernel="${BUILD_KERNEL}" >/dev/null 2>&1; echo $$?)
 endif # check_aux_bus exists
 
 # The out-of-tree auxiliary module we ship should be moved into this
-- 
2.42.0

